### DISCUSSION TITLE
option to enable continuous showing pages *horizontally*

### DISCUSSION THREAD

**Discussion by enihcam** [sumatrapdfreader/sumatrapdf#3023] [Ideas]
Yes, right now we have the option `Show Pages Continuously`, but it is continuous vertically.
To better mimic the real book-reading experience, i would suggest one more option for horizontal page continuation.

e.g.
`[253 254] 255 256` -> `253 [254 255] 256`  // expected behavior, better with animation
NOT
`[253 254] 255 256` -> `253 254 [255 256]`  // current behavior, no animation

Thanks!

-----Note-----
This is different from https://github.com/sumatrapdfreader/sumatrapdf/discussions/2463 which asks for horizontal scrolling. Instead, I'm only asking for horizontal page continuation, fake-scrolling animation is good enough.

**Comment by nicohofdtz** [comment]
I would like to second this. It would be particularly helpful when reading book that feature figures and tables that are referenced in the text as those are often placed on neigboughring sides.

**Reply by GitHubRulesOK** [reply]
@nicohofdtz 
for pairing pages to left or right you need to flip "book" and "facing" (ctrl 7&8) as required by the authored layout, this is different to the request for single page continuation (which is not a trivial task even if result would be the same)

**Reply by nicohofdtz** [reply]
Thanks for pointing me to that options. This might work as a workaround. However, it just toggles if even pages are on the left or on the right side and "next page" always advances by two pages. What I would want is a viewing option where two (or more -> page continuation) pages are displayed next to one another and "next page" only advances one page at a time. That's exactly what "Show pages continously" does right now except it only works vertically.

**Comment by meteorquake** [comment]
Slightly more expanding this, it would be useful at times to tile as many pages as fit in a screen, or a specified grid n1 x n1, scrolling down to see more, although I suspect not showing part pages at the bottom would make the task simpler. I presume this would be just a grid of what Sumatra already shows. There could be an offset so that it falls in the way the user wants it. The original question would be fulfilled by a particular setting of this where the grid is 2 x 1 and the offset is stepped down by 1 to slide the pages horizontally forward, and up by 1 for sliding backward.
d

**Comment by kjohn740** [comment]
I would like to second this as well. It would be nice to be able to scroll just one page at the time in two-page view.

**Comment by T1mL3arn** [comment]
Would like to see this page view setting too. In PDF-XChangeEditor it is called "Ribbon Layout".

<details>
  <summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/be533168-3740-425a-b456-f6e53fb4491f)

</details>

**Comment by meteorquake** [comment]
Scrolling horizontally is a great idea, since screens are often quite wide now and can fit 3-4 pages across, horizontal scrolling makes intuitive sense.
I dare say if it were available it would be my most commonly used layout to fill the screen horizontally with pages (1 page high) and scroll horizontally. (I can imagine more niche cases where you might want to view across with 2+ pages high.)
Then when I want to focus, a single page view would probably help. Occasionally 2+ pages are useful for focus.
I think such focus cases could use the same layout of filling the screen horizontally but the user could specify the number of pages and the extra would be dimmed. So if I had 4 pages across and wanted to see just 2 it would be sufficient to show the 2 in the centre with the remainder dimmed (perhaps a user dimming setting), and scrolling would be horizontal bringing the dimmed pages across into the normal brightness area.

**Comment by GitHubRulesOK** [comment]
PDF pages are actually not always in a fixed order, often they are randomly spread throughout a document.

Thus a PDF or eBook archive are then assembled into a rational layout. Which grid like can be one beneath the other in both columns and or in rows.

columns  of one or 2 pages are the way most readers will place the ordered contiguous contents
1 2
3 4
5 6
or
   1
2 3
4 5
6
or 
1 3 5
2 4 6
or as asked here
1 2 3
4 5 6

these are all covered by the open issue https://github.com/sumatrapdfreader/sumatrapdf/issues/4184
